File: intermediate-server/factory.intermediate.iot/src/main/java/ua/nure/makieiev/factory/intermediate/iot/service/FilterWorkLogService.java
package ua.nure.makieiev.factory.intermediate.iot.service;

import ua.nure.makieiev.factory.intermediate.iot.model.FilterUnit;
import ua.nure.makieiev.factory.intermediate.iot.model.FilterWorkLogInfo;

public interface FilterWorkLogService {

    boolean save(FilterWorkLogInfo filterWorkLogInfo, String token);

    FilterUnit findFilterUnitByUnitId(long unitId, String token);

}

File: backend/src/main/kotlin/ua/nure/makieiev/ark/service/FilterWorkLogService.kt
package ua.nure.makieiev.ark.service

import ua.nure.makieiev.ark.model.entity.FilterWorkLog
import ua.nure.makieiev.ark.model.entity.FilterWorkLogStatistic
import java.util.*

interface FilterWorkLogService {

    fun save(filterWorkLog: FilterWorkLog): FilterWorkLog

    fun findById(id: Long): Optional<FilterWorkLog>

    fun findAll(): List<FilterWorkLog>

    fun findAllByFilterUnitId(filterUnitId: Long): List<FilterWorkLog>

    fun getStatisticByFilerWorkLog(): List<FilterWorkLogStatistic>

}